### PR TITLE
Fix broken job specs

### LINKED ISSUE #4269: Some job tests are not being executed and they do not fail
**Describe the bug**

Tests inside a ``perform_now`` scope are never executed. Take, for example, ``bust_cache_job_spec.rb``:

https://github.com/thepracticaldev/dev.to/blob/66faaccdd894a75887e6e3dfad30a3152e18a4fb/spec/jobs/articles/bust_cache_job_spec.rb#L20-L22

This line has a typo. It's supposed to be have_receive**d**, and yet the test passes. The reason for that is because the test is inside a ``describled_class.perform_now`` scope, which is never executed. 

**To Reproduce**

Change the test to:

```ruby
it "async busts cache" do
  cache_buster = double
  allow(cache_buster).to receive(:bust_article)

  described_class.perform_now(article.id, cache_buster)
  expect(cache_buster).to have_receive(:bust_article).with(article)
end
```
and it will fail because of the typo in ``have_receive``.

Another way to see that this is never executed is using ``pry``. Add a ``binding.pry`` inside the ``perform_now`` scope and run the test - the ``pry`` is never reached. 

**Expected behavior**

There should not be any ``perform_now`` scopes in tests, because nothing inside those get executed. 

**Additional context**

These are the tests that contain a scope like that:

- [ ] articles/bust_cache_job_spec.rb
- [ ] detect_human_language_job_spec.rb
- [ ] score_calc_job_spec.rb
- [ ] update_main_image_background_hex_job_spec.rb
- [ ] calculate_score_job_spec.rb
- [ ] create_id_code_job_spec.rb
- [ ] send_email_notification_job_spec.rb
- [ ] touch_user_job_spec.rb
- [ ] populate_history_job_spec.rb
- [ ] users/bust_cache_job_spec.rb
- [ ] resave_articles_job_spec.rb

### PR DESCRIPTION
**Pull request by austenmadden**
Resolves #4269 

## What type of PR is this? (check all applicable)

- [x] Refactor
- [ ] Feature
- [x] Bug Fix
- [ ] Documentation Update

## Description

Many job specs were putting their assertions inside a block provided to
ActiveJob's perform_now. Unfortunately, this syntax does not error and
simply ignores the provided block.

The tests were therefore not executing their assertions. As a result
many of them were actually broken to begin with. Many used incorrect
stubbing, were trying to stub instances of models that wouldn't actually
be returned by ActiveRecord queries, etc.

This commit fixes the syntax and does it's best to fix the tests. Many
of these assertions/spec could use more TLC, but this work at least gets
them passing/executing.

## Related Tickets & Documents
#4269 

## Added to documentation?

- [ ] docs.dev.to
- [ ] readme
- [x] no documentation needed

### CODE REVIEW

File: spec/jobs/comments/send_email_notification_job_spec.rb [hunk 1]
```diff
@@ -6,11 +6,10 @@
     let(:comment) { FactoryBot.create(:comment, commentable: article) }
 
     it "sends notify email" do
-      allow(NotifyMailer).to receive(:new_reply_email)
+      allow(NotifyMailer).to receive(:new_reply_email).and_call_original
```
Comment (lightalloy) [inline spec/jobs/comments/send_email_notification_job_spec.rb]: Do we really need to `call_original` here?

File: spec/jobs/users/subscribe_to_mailchimp_newsletter_job_spec.rb [hunk 2]
```diff
@@ -7,13 +7,12 @@
     let(:user) { FactoryBot.create(:user) }
 
     it "subscribes user to mailchimp newsletter" do
-      mailchimp_bot = MailchimpBot.new(user)
+      mailchimp_bot = double
+      allow(MailchimpBot).to receive(:new).and_return(mailchimp_bot)
 
-      allow(mailchimp_bot).to receive(:upsert)
+      expect(mailchimp_bot).to receive(:upsert)
```
Comment (lightalloy) [inline spec/jobs/users/subscribe_to_mailchimp_newsletter_job_spec.rb]: I suggest adding `allow(mailchimp_bot).to receive(:upsert)`, changing expectation to `have_received`, and moving `expect` lower than `perform_now`. It's easier to read and understand like this, and `rubocop-rspec` recommends this.

Comment (austenmadden) [reply spec/jobs/users/subscribe_to_mailchimp_newsletter_job_spec.rb]: Out of curiosity, why doesn't it run during the build? Would save some review time.

File: spec/jobs/articles/score_calc_job_spec.rb [hunk 3]
```diff
@@ -7,24 +7,31 @@
...
+      expect(article.spaminess_rating).to be(2)
     end
 
-    it "does not update article scores when no article" do
-      described_class.perform_now(article.id) do
-        expect(article.score).not_to be(7)
-        expect(article.hotness_score).not_to be(373)
-        expect(article.spaminess_rating).not_to be(2)
+    context "without article" do
+      let(:article) { nil }
```
Comment (lightalloy) [inline spec/jobs/articles/score_calc_job_spec.rb]: Same as above, `article` is not used.

File: spec/jobs/articles/update_main_image_background_hex_job_spec.rb [hunk 4]
```diff
@@ -7,18 +7,17 @@
...
-      end
+      described_class.perform_now(article.id)
+      expect(article.reload.main_image_background_hex_color).to eql("#eee")
     end
 
-    it "does not update articles main image background hex when no article is found" do
-      color_from_image = double
-      allow(color_from_image).to receive(:main).and_return("#eee")
+    context "without article" do
+      let(:article) { nil }
```
Comment (lightalloy) [inline spec/jobs/articles/update_main_image_background_hex_job_spec.rb]: Same as above.

File: spec/jobs/users/subscribe_to_mailchimp_newsletter_job_spec.rb (hunk 2 above)
Comment (lightalloy) [reply spec/jobs/users/subscribe_to_mailchimp_newsletter_job_spec.rb]: Thanks for raising this question. I checked in with the team and it's like this only for historical reasons. We'll work on enabling it.

**Comment by rhymes** [comment]
I submit a PR - https://github.com/thepracticaldev/dev.to/pull/4334 - to deal with the unrelated test failing, i'm going to clear the cache and restart the build

**Pull request merged**